experiment/scheduler.py

```diff
@@ -701,6 +701,7 @@ def __init__(self, trial):
         self.time_ended = trial.time_ended
         self.preemptible = trial.preemptible
         self.cpuset = None
+        self.trial_group_num = trial.trial_group_num
```

Comment:
I think this has been plumbed through properly but I'm not 100% sure (no reason to think it's wrong). Sorry this setup isn't great.

Reply:
Hi @jonathanmetzman the `trial_group_num` field is used as an identifier of the group of trials and this idea of trial grouping is used to make sure that we provide the same set of seeds to certain trails (within the same group).

For example, if in the config.yaml we set trials to 2 for the experiment so we have following trials in total (for fuzzer afl, libfuzzer and benchmark libjpeg):
- trial_id: 1, benchmark: libjpeg, fuzzer: afl
- trial_id: 2, benchmark: libjpeg, fuzzer: afl
- trial_id: 3, benchmark: libjpeg, fuzzer: libfuzzer
- trial_id: 4, benchmark: libjpeg, fuzzer: libfuzzer

so fuzzbench will generate following group (synthetically) and assign the same set initial seeds to trials within the same group:
- trial_id: 1, benchmark: libjpeg, fuzzer: afl   --> Group A
- trial_id: 2, benchmark: libjpeg, fuzzer: afl   --> Group B
- trial_id: 3, benchmark: libjpeg, fuzzer: libfuzzer --> Group A
- trial_id: 4, benchmark: libjpeg, fuzzer: libfuzzer --> Group B

Please let me know if it makes sense? also is there a way to only add/inject this `trial_group_num` field only when are running in `MICRO_EXPERIMENT` mode?